app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -447,29 +447,46 @@ export default SiteHeaderComponent.extend({
 
   init() {
     this._super(...arguments);
-
     this._resizeObserver = null;
   },
 
+  @bind
+  updateHeaderOffset() {
+    // check offset from top of the page, in case there's a custom header above
+    // and adjust on scroll in case it's not sticky
+    if (this.scrollAnimationFrame) {
+      cancelAnimationFrame(this.scrollAnimationFrame);
+    }
+
+    this.scrollAnimationFrame = requestAnimationFrame(() => {
```

Comment:
Do you remember the reason for adding `requestAnimationFrame` here @awesomerobot